stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml

```diff
@@ -15,52 +15,48 @@ spec:
 {{ include "gcpdisk.labels" . | indent 6 }}
         app: gcp-compute-persistent-disk-csi-driver
     spec:
-      # Host network must be used for interaction with Workload Identity in GKE
-      # since it replaces GCE Metadata Server with GKE Metadata Server. Remove
-      # this requirement when issue is resolved and before any exposure of
-      # metrics ports
-      hostNetwork: true
       serviceAccountName: csi-gce-pd-controller-sa
       nodeSelector:
         beta.kubernetes.io/os: linux
       priorityClassName: system-cluster-critical
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
       containers:
         - name: csi-provisioner
           image: {{ .Values.provisioner.image.repository }}:{{ .Values.provisioner.image.tag }}
           args:
+            - --v=5
             - --csi-address=$(ADDRESS)
+            - --metrics-address=:22011
             - --timeout={{ .Values.provisioner.timeout }}
             - --retry-interval-start={{ .Values.provisioner.retryIntervalStart }}
             - --retry-interval-max={{ .Values.provisioner.retryIntervalMax }}
             - --worker-threads={{ .Values.provisioner.workerThreads }}
             {{- if .Values.enableTopologyZone }}
             - --feature-gates=Topology=true
             {{- end}}
+            {{- if (gt (.Values.replicas | int) 1) }}
             - --enable-leader-election
             - --leader-election-type=leases
+            {{- end }}
           env:
             - name: ADDRESS
               value: /csi/csi.sock
           imagePullPolicy: Always
           volumeMounts:
             - mountPath: /csi
               name: socket-dir
-        - name: cluster-driver-registrar
-          image: {{ .Values.registrar.cluster.image.repository }}:{{ .Values.registrar.cluster.image.tag }}
-          args:
-            - --csi-address=$(ADDRESS)
-            - --driver-requires-attachment=true
-            - --v=5
-          env:
-            - name: ADDRESS
-              value: /csi/csi.sock
-          volumeMounts:
-            - name: socket-dir
-              mountPath: /csi
         - name: csi-snapshotter
-          image: {{ .Values.snapshotter.image.repository }}:{{ .Values.snapshotter.image.tag }}
+          image: "{{ .Values.snapshotter.image.repository }}:{{ include "gcpdisk.snapshotter.image.tag" . }}"
           args:
+            - --v=5
             - --csi-address=$(ADDRESS)
+            - --metrics-address=:22014
```

Comment:
are these ports the default ones ? why to set them then ?

Reply:
`--metrics-address: The TCP network address where the prometheus metrics endpoint will run (example: :8080 which corresponds to port 8080 on local host). The default is empty string, which means metrics endpoint is disabled.`